### DISCUSSION TITLE
How to add simple JUnit tests to m2e module

### DISCUSSION THREAD

**Discussion by kwin** [eclipse-m2e/m2e-core#1165] [General]
Tycho perfectly supports tests with [JUnit running inside Eclipse](https://tycho.eclipseprojects.io/doc/latest/tycho-surefire-plugin/plugin-info.html). I am wondering, though what is the best approach for plain JUnit test cases which don't require an OSGi runtime. Several options come to my mind:

a) Adding the test cases in the same PDE project as the actual code to test and execute that via Tycho with an explicit call to maven-surefire-plugin
b) Adding the test cases in another dedicated test project which is a simple Maven module (no Eclipse specific metadata necessary) with packaging `jar` which automatically executes those through [their default binding](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#default-lifecycle-bindings-packaging-ejb-ejb3-jar-par-rar-war)
c) Just execute those tests inside Eclipse similar to the other ITs (although unnecessary overhead for those)

Which one would you guys prefer for a test for code in https://github.com/eclipse-m2e/m2e-core/tree/master/org.eclipse.m2e.launching?

**Comment by laeubi** [comment]
> Tycho perfectly supports tests with [JUnit running inside Eclipse](https://tycho.eclipseprojects.io/doc/latest/tycho-surefire-plugin/plugin-info.html).

Actually Tycho do not do anything inside Eclipse, do you mean running eclipse as part of the test? Or "Pugin Tests" (with or w/o UI)

> I am wondering, though what is the best approach for plain JUnit test cases which don't require an OSGi runtime. 

The usual way here would be to create a test-source-folder in the plugin and just use surefire.
